src/preferences/credentials.cpp

```diff
@@ -245,15 +246,24 @@ namespace preferences
 		}
 		try {
 			filesystem::scoped_ostream credentials_file = filesystem::ostream_file(filesystem::get_credentials_file());
-			secure_buffer encrypted = encrypt(credentials_data, build_key("global", get_system_username()));
+			secure_buffer encrypted = aes_encrypt(credentials_data, build_key("global", get_system_username()));
 			credentials_file->write(reinterpret_cast<const char*>(encrypted.data()), encrypted.size());
 		} catch(const filesystem::io_exception&) {
 			ERR_CFG << "error writing to credentials file '" << filesystem::get_credentials_file() << "'";
 		}
 	}
 }
 
-// TODO: Key-stretching (bcrypt was recommended)
+/**
+ * Fills a secure_buffer with 32 bytes of deterministically generated bytes, then overwrites it with the system login name, server login name, and server name.
+ * If this is more than 32 bytes, then it's truncated. If it's less than 32 bytes, then the pre-generated bytes are used to pad it.
+ *
+ * TODO: Key-stretching (bcrypt was recommended)
```

Comment:
Does this TODO still make sense in the code, now we've already accepted that it's a trivial (hardcoded key) encryption?

Reply:
Not really, IMO.